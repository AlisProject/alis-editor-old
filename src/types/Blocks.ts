export enum BlockType {
  Rule = 'Rule',
  Text = 'Text',
  Paragraph = 'Paragraph',
  Image = 'Image',
  Quote = 'Quote',
  Heading = 'Heading',
  Embed = 'Embed'
}

export interface Block {
  id: string
  type: BlockType
  payload?: any
  children?: Block[]
}

export interface RuleBlock extends Block {
  type: BlockType.Rule
}

export interface EmbedBlock extends Block {
  type: BlockType.Embed,
  payload: {
    src: string
  }
}

export interface ParagraphBlock extends Block {
  type: BlockType.Paragraph
  payload: {
    body: string
  }
}

export interface TextBlock extends Block {
  type: BlockType.Text
  payload: {
    body: string
  }
}

export interface ImageBlock extends Block {
  type: BlockType.Image
  payload: {
    src: string
    align?: string
  }
  children: Block[]
}

export interface HeadingBlock extends Block {
  type: BlockType.Heading
  payload: {
    size: 'h2' | 'h3'
    body: string
  }
  children: Block[]
}

export interface QuoteBlock extends Block {
  type: BlockType.Quote
  children: Block[]
}

/*
「詳細は [こちら](https://example.com) から確認できます」を再現する場合
[
  {
    type: 'Paragraph',
    children: [
      {
        type: 'Text',
        payload: {
          body: '詳細は'
        }
      },
      {
        type: 'Link',
        payload: {
          href: 'https://example.com'
        },
        children: [
          {
            type: 'Text',
            payload: {
              body: 'こちら'
            }
          }
        ]
      },
      {
        type: 'Text',
        payload: {
          body: 'から確認できます'
        }
      }
    ]
  }
]

*/
