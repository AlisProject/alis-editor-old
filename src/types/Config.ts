export interface AlisEditorConfig {
  articleId: string
  preview: boolean
  iframelyApikey: string
  uploadEndpoint: string
}
