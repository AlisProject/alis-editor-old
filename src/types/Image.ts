export interface ImageUploadRequest {
  article_image: string
}

export interface ImageUploadResponse {
  image_url: string
}
