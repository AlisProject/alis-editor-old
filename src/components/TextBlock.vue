<template>
  <textarea
    @click="handleFocus"
    @input="handleInput"
    @keydown="handleKeydown"
    :rows="rows"
    :data-id="typedBlock.id"
    :placeholder="placeholder || ''"
  >{{typedBlock.payload.body}}</textarea>
</template>

<script lang="ts">
import Vue from 'vue'
import { TextBlock } from '../types/Blocks'

export default Vue.extend({
  props: {
    block: Object,
    placeholder: String
  },
  computed: {
    rows() {
      return (this as any).typedBlock.payload.body.split('\n').length
    },
    typedBlock(): TextBlock {
      return this.block
    }
  },
  methods: {
    handleFocus() {
      this.$emit('focus')
    },
    handleKeydown(event: KeyboardEvent) {
      if (event.keyCode !== 8 || this.typedBlock.payload.body) {
        return
      }
      this.$emit('delete')
    },
    handleInput(event: KeyboardEvent) {
      console.log('hoge')
      const value = (event.target! as HTMLTextAreaElement).value
      this.$emit('input', value)
    }
  }
})
</script>

<style scoped>
textarea {
  border: 0;
  outline: none;
  resize: none;
  width: 100%;
  line-height: 1.65;
  overflow: hidden;
  background: transparent;
  padding: 0 8px;
}
</style>
