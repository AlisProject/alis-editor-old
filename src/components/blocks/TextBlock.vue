<template>
  <textarea
    @click="handleFocus"
    @input="handleInput"
    @keydown="handleKeydown"
    :rows="rows"
    :data-id="block.id"
    :placeholder="placeholder || ''"
    v-if="!preview"
  >{{block.payload.body}}</textarea>
  <span
    v-else
    class="text preview"
  >
    {{block.payload.body}}
  </span>
</template>

<script lang="ts">
import Vue from 'vue'
import { TextBlock, BlockType } from '../../types/Blocks'

export default Vue.extend({
  props: {
    block: Object as () => TextBlock,
    placeholder: String,
    preview: {
      default: false,
      type: Boolean
    }
  },
  computed: {
    rows() {
      return (this as any).block.payload.body.split('\n').length
    }
  },
  methods: {
    handleFocus() {
      this.$emit('focus')
    },
    handleKeydown(event: KeyboardEvent) {
      if (event.keyCode !== 8 || this.block.payload.body) {
        return
      }
      this.$emit('delete', this.block)
    },
    handleInput(event: KeyboardEvent) {
      const value = (event.target! as HTMLTextAreaElement).value
      this.$emit('input', value)
    }
  }
})
</script>

<style scoped>
textarea {
  border: 0;
  outline: none;
  resize: none;
  width: 100%;
  line-height: 1.65;
  overflow: hidden;
  background: transparent;
  padding: 0 8px;
}

.text.preview {
  line-height: 1.65;
  padding: 0 8px;
}
</style>
