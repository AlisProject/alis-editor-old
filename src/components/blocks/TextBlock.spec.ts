import { mount, config } from '@vue/test-utils'
import TextBlock from './TextBlock.vue'
import * as BlockTypes from '../../types/Blocks'
config.logModifiedComponents = false

describe('TextBlock.vue', () => {
  test('renders correctly', () => {
    const block: BlockTypes.TextBlock = {
      id: '123',
      type: BlockTypes.BlockType.Text,
      payload: {
        body: 'foo'
      }
    }
    const wrapper = mount(TextBlock, {
      propsData: {
        block
      },
      stubs: ['ShadowInput']
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders correctly (preview mode)', () => {
    const block: BlockTypes.TextBlock = {
      id: '123',
      type: BlockTypes.BlockType.Text,
      payload: {
        body: 'foo'
      }
    }
    const wrapper = mount(TextBlock, {
      propsData: {
        block,
        preview: true
      },
      stubs: ['ShadowInput']
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
