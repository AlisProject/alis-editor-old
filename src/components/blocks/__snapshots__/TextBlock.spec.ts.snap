// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock.vue renders correctly (preview mode) 1`] = `
<span
  class="text preview"
>
  
  foo

</span>
`;

exports[`TextBlock.vue renders correctly 1`] = `
<textarea
  data-id="123"
  placeholder=""
  rows="1"
>
  foo
</textarea>
`;
